src/proto/h2/mod.rs

```diff
@@ -13,13 +13,30 @@ pub(crate) use self::client::Client;
 pub(crate) use self::server::Server;
 
 fn strip_connection_headers(headers: &mut HeaderMap) {
-    if headers.remove(TRANSFER_ENCODING).is_some() {
-        trace!("removed illegal Transfer-Encoding header");
-    }
-    if headers.contains_key(CONNECTION) {
-        warn!("Connection header illegal in HTTP/2");
-        //TODO: actually remove it, after checking the value
-        //and removing all related headers
+    let connection_headers = vec![
```

Comment:
Make this an array, maybe?

Reply:
:+1: That thought completely skipped my mind